packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -312,6 +312,13 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
   // Handles up the logic for dragging and reordering items in the list.
   Widget _wrap(Widget toWrap, int index, BoxConstraints constraints) {
```

Comment:
If I provide a list of three stateful children with keys A, B, and C then if I rebuild the ReorderableList with the same children ordered C, B , A then I'd expect  that the state of C, B, and A will have been preserved. Because they appear to be siblings, even though they're not actually siblings. In other words, I would not expect any of the  the A, B, C widgets' state to be disposed.

Since each GlobalKey includes the child's index, I don't think child state will be preserved if the ReorderableList is rebuilt with the same children, but in a different order.

We  should probably have a test that checked this.

Reply:
Done.